bin/gce/google_cloud_mango_install.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -e
+
+ROLE=$(curl -f -s -H Metadata-Flavor:Google http://metadata/computeMetadata/v1/instance/attributes/dataproc-role)
+INIT_ACTIONS_REPO=$(curl -f -s -H Metadata-Flavor:Google http://metadata/computeMetadata/v1/instance/attributes/INIT_ACTIONS_REPO || true)
+INIT_ACTIONS_REPO="${INIT_ACTIONS_REPO:-https://github.com/GoogleCloudPlatform/dataproc-initialization-actions.git}"
+INIT_ACTIONS_BRANCH=$(curl -f -s -H Metadata-Flavor:Google http://metadata/computeMetadata/v1/instance/attributes/INIT_ACTIONS_BRANCH || true)
+INIT_ACTIONS_BRANCH="${INIT_ACTIONS_BRANCH:-master}"
+DATAPROC_BUCKET=$(curl -f -s -H Metadata-Flavor:Google http://metadata/computeMetadata/v1/instance/attributes/dataproc-bucket)
+
+# Colon-separated list of conda channels to add before installing packages
+JUPYTER_CONDA_CHANNELS=$(curl -f -s -H Metadata-Flavor:Google http://metadata/computeMetadata/v1/instance/attributes/JUPYTER_CONDA_CHANNELS || true)
+# Colon-separated list of conda packages to install, for example 'numpy:pandas'
+JUPYTER_CONDA_PACKAGES=$(curl -f -s -H Metadata-Flavor:Google http://metadata/computeMetadata/v1/instance/attributes/JUPYTER_CONDA_PACKAGES || true)
+
+echo "Cloning fresh dataproc-initialization-actions from repo $INIT_ACTIONS_REPO and branch $INIT_ACTIONS_BRANCH..."
+git clone -b "$INIT_ACTIONS_BRANCH" --single-branch $INIT_ACTIONS_REPO
+# Ensure we have conda installed.
+./dataproc-initialization-actions/conda/bootstrap-conda.sh
+
+source /etc/profile.d/conda.sh
+
+if [ -n "${JUPYTER_CONDA_CHANNELS}" ]; then
+  echo "Adding custom conda channels '$(echo ${JUPYTER_CONDA_CHANNELS} | tr ':' ' ')'"
+  conda config --add channels $(echo ${JUPYTER_CONDA_CHANNELS} | tr ':' ',')
+fi
+
+if [ -n "${JUPYTER_CONDA_PACKAGES}" ]; then
+  echo "Installing custom conda packages '$(echo ${JUPYTER_CONDA_PACKAGES} | tr ':' ' ')'"
+  conda install $(echo ${JUPYTER_CONDA_PACKAGES} | tr ':' ' ')
+fi
+
+if [[ "${ROLE}" == 'Master' ]]; then
+    conda install jupyter
+    if gsutil -q stat "gs://$DATAPROC_BUCKET/notebooks/**"; then
+        echo "Pulling notebooks directory to cluster master node..."
+        gsutil -m cp -r gs://$DATAPROC_BUCKET/notebooks /root/
+    fi
+    ./dataproc-initialization-actions/jupyter/internal/setup-jupyter-kernel.sh
+    ./dataproc-initialization-actions/jupyter/internal/launch-jupyter-kernel.sh
+    # Install docker
+    wget -O - -q 'https://github.com/bigdatagenomics/mango/blob/master/bin/gce/google_cloud_docker_install.sh' | sudo bash
+fi
+echo "Completed installing Jupyter!"
+
+# Install Jupyter extensions (if desired)
+# TODO: document this in readme
+if [[ ! -v $INSTALL_JUPYTER_EXT ]]
+    then
+    INSTALL_JUPYTER_EXT=false
+fi
+if [[ "$INSTALL_JUPYTER_EXT" = true ]]
+then
+    echo "Installing Jupyter Notebook extensions..."
+    ./dataproc-initialization-actions/jupyter/internal/bootstrap-jupyter-ext.sh
+    echo "Jupyter Notebook extensions installed!"
+fi
+
+pip install cigar
```

Comment:
shouldn't cigar be installed in the virtual env?

Reply:
This is so cigar is present on the worker nodes (it is also installed in the venv).

What configurations have you been setting to port python packages over to the worker nodes on the cluster? It looks like the python binary for the worker nodes is not being overwritten by the flags sent to Spark, which leads to "cigar not found" issues if it is not installed this way.

Reply:
won't this be installed outside of the virtual env? Do you know why cigar is causing an issue and not other imports?

Reply:
Cigar will be installed in the virtual env, just not through this process.

Since the pyspark jobs depend on cigar in the worker nodes, the python distribution on the worker nodes need to have cigar present. I don't believe we use any other imports in pyspark for mango-notebook.